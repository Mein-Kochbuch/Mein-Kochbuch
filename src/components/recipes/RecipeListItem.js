import React from "react";
import styled from "styled-components/native";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";

export default function RecipeListItem({recipe}) {
    const isDarkMode = useColorScheme() === 'dark';

    const imageSource = recipe.thumbnail_url ? {uri: recipe.thumbnail_url} : require("../../../resources/platzhalter.png")

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode}>
            <ImageStyled source={imageSource}/>
            <StyledTest numberOfLines={2} ellipsizeMode={"middle"}>
                {recipe.title}
            </StyledTest>
        </StyledTouchableOpacity>
    )
}

const ImageStyled = styled.Image`
  height: 100px;
  width: 100px;
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`

const StyledTest = styled.Text`
  text-align: center;
  font-size: 24px;
  margin-left: 12px;
  max-width: 70%;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  padding: 12px;
  margin: 12px;
  height: 120px;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
