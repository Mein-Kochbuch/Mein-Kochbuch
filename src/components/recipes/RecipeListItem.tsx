import React from 'react';
import styled from 'styled-components/native';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import {Appearance, ListRenderItemInfo} from 'react-native';
import {useHistory} from 'react-router-native';
import {RecipePreview} from '../../models/RecipePreview';

export type RecipeListItemProps = {
  listItem: ListRenderItemInfo<RecipePreview>;
};

export default function RecipeListItem({listItem}: RecipeListItemProps) {
  const history = useHistory();
  const isDarkMode = Appearance.getColorScheme() === 'dark';

  const recipe = listItem.item;
  const imageSource = recipe.thumbnail?.url
    ? {uri: recipe.thumbnail.url}
    : require('../../../resources/platzhalter.png');

  const handleOnPress = () => {
    history.push('/recipes/' + recipe.id);
  };

  return (
    <StyledTouchableOpacity isDarkMode={isDarkMode} onPress={handleOnPress}>
      <ImageStyled source={imageSource} />
      <StyledTest numberOfLines={2} ellipsizeMode={'middle'}>
        {recipe.title}
      </StyledTest>
    </StyledTouchableOpacity>
  );
}

const ImageStyled = styled.Image`
  height: 100px;
  width: 100px;
  padding-top: 20px;
  padding-bottom: 20px;
  background-color: #68a0cf;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`;

const StyledTest = styled.Text`
  text-align: center;
  font-size: 24px;
  margin-left: 12px;
  max-width: 70%;
`;

interface StyledTouchableOpacityProps {
  isDarkMode: boolean;
}

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableOpacityProps>`
  padding: 12px;
  margin: 12px;
  height: 120px;
  display: flex;
  flex-direction: row;
  align-items: center;
  border-style: solid;
  border-color: ${props => (props.isDarkMode ? Colors.lighter : Colors.darker)};
  border-radius: 10px;
  border-width: 2px;
`;
