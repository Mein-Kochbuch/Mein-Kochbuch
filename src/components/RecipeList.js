import React from "react";
import {View} from "react-native";
import RecipeListItem from "./RecipeListItem";

export default function RecipeList({recipes, filter}) {

    return (
        <View>
            {recipes.map(recipe => <RecipeListItem key={recipe.pk} recipe={recipe}/>)}
        </View>
    )
}
