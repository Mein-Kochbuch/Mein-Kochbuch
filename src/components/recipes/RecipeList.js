import React from "react";
import {View} from "react-native";
import RecipeListItem from "./RecipeListItem";

export default function RecipeList({recipes}) {

    return (
        <View>
            {recipes.map(recipe => <RecipeListItem key={recipe.pk} recipe={recipe}/>)}
        </View>
    )
}
