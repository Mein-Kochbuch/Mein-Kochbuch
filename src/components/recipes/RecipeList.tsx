import React from 'react';
import {FlatList} from 'react-native';
import RecipeListItem from './RecipeListItem';
import styled from 'styled-components/native';
import {RecipePreview} from '../../models/RecipePreview';

interface RecipeListProps {
  recipes: RecipePreview[];
  loadMoreRecipes: () => void;
}

export default function RecipeList({
  recipes,
  loadMoreRecipes,
}: RecipeListProps) {
  return (
    <ViewStyled>
      <FlatList
        data={recipes}
        renderItem={item => <RecipeListItem item={item} />}
        keyExtractor={item => item.id}
        onEndReached={loadMoreRecipes}
      />
    </ViewStyled>
  );
}

const ViewStyled = styled.View`
  height: 88%;
`;
