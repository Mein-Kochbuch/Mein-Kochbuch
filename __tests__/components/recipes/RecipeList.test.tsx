import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import RecipeList from '../../../src/components/recipes/RecipeList';
import RecipeListItem from '../../../src/components/recipes/RecipeListItem';

jest.mock('../../../src/components/recipes/RecipeListItem', () => {
  return props => {
    return <div {...props} />;
  };
});

const recipes = [
  {id: '1', title: 'test-title-1'},
  {id: '2', title: 'test-title-2'},
];

it('RecipeList Test', () => {
  const component = renderer.create(<RecipeList recipes={recipes} />);

  expect(component.toJSON()).toMatchSnapshot();
  expect(component.toJSON().children.length).toBe(1);

  const testInstance = component.root;

  expect(
    testInstance.findAllByType(RecipeListItem)[0].props.item.item,
  ).toStrictEqual({
    title: 'test-title-1',
    id: '1',
  });

  expect(
    testInstance.findAllByType(RecipeListItem)[1].props.item.item,
  ).toStrictEqual({
    title: 'test-title-2',
    id: '2',
  });
});
