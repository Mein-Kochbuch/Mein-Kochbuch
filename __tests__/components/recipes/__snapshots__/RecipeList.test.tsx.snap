// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList Test 1`] = `
<View
  style={
    Array [
      Object {
        "height": "88%",
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "1",
          "title": "test-title-1",
        },
        Object {
          "id": "2",
          "title": "test-title-2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <div
          item={
            Object {
              "index": 0,
              "item": Object {
                "id": "1",
                "title": "test-title-1",
              },
              "separators": Object {
                "highlight": [Function],
                "unhighlight": [Function],
                "updateProps": [Function],
              },
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <div
          item={
            Object {
              "index": 1,
              "item": Object {
                "id": "2",
                "title": "test-title-2",
              },
              "separators": Object {
                "highlight": [Function],
                "unhighlight": [Function],
                "updateProps": [Function],
              },
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
